src/main/java/org/eclipse/yasson/internal/AnnotationFinder.java

```diff
@@ -0,0 +1,133 @@
+package org.eclipse.yasson.internal;
+
+import org.eclipse.yasson.internal.properties.MessageKeys;
+import org.eclipse.yasson.internal.properties.Messages;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+
+/**
+ * Finds an annotation including inherited annotations (e.g. meta-annotations).
+ * 
+ * @author JohT
+ */
+class AnnotationFinder {
+
+    private static final String CONSTRUCTOR_PROPERTIES_ANNOTATION = "java.beans.ConstructorProperties";
+    private static final Logger LOGGER = Logger.getLogger(AnnotationFinder.class.getName());
+
+    private final String annotationClassName;
+    private final Class<? extends Annotation> annotationClass; // may be null
+
+    /**
+     * Gets the {@link AnnotationFinder} for the given Annotation-Type.
+     * 
+     * @param annotation {@link Class}, that is a sub-type of {@link Annotation}
+     * @return {@link AnnotationFinder}
+     */
+    public static final AnnotationFinder findAnnotation(Class<?> annotation) {
```

Comment:
`AnnotationFinder` should be "manually injected" inside `AnnotationIntrospector`, similar to other dependencies during `JsonbContext` creation. That way it will be in the same fashion and avoid static calls.

Reply:
Since i use the annotation name (and type) as final properties in AnnotationFinder, i would need to "manually inject" a couple of them, e.g.:
AnnotationFinder jsonbDateFormatAnnotationFinder = AnnotationFinder.findAnnotation(JsonbDateFormat.class);

I could get rid of the private final properties in AnnotationFinder, but that would be nearly the same as the current static solution (more functional, less object-oriented).

I could replace the implementation of findAnnotation by
return AnnotationFinder.findAnnotation(annotationClass).in(declaredAnnotations);
but that would create a new object each time it is called.

Since AnnotationFinder is immutable and uses static factory methods i could build in caching. I already tried that way out and deleted it, since it introduced complexity that does not felt right  for me.

I'll thing about it, maybe there is another way....

Reply:
I missed that it stores processing parameters internally. Creating new instance frequently is not hitting performance with modern JDKs and is not considered a bad practice.